### PR TITLE
V2 conformal evaluators

### PR DESCRIPTION
**Pull request by shihchengli**
## Description
Implement of conformal evaluators:

- MulticlassConformalEvaluator
- MultilabelConformalEvaluator
- RegressionConformalEvaluator

## Checklist
- [x] linted with flake8?
- [x] (if appropriate) unit tests added?

### CODE REVIEW

File: chemprop/uncertainty/evaluator.py [hunk 1]
```diff
@@ -73,21 +74,77 @@ def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -
...
+            targets_j = targets[:, j][mask_j]
+            interval_j = uncs[:, j][mask_j]
+            bounds = torch.tensor([-1 / 2, 1 / 2]).view(-1, 1)
+            interval_bounds = bounds * interval_j.unsqueeze(0)
+            pred_bounds = preds_j.unsqueeze(0) + interval_bounds
+            task_results = torch.logical_and(
+                pred_bounds[0, :] <= targets_j, targets_j <= pred_bounds[1, :]
+            )
+            coverages.append(task_results.sum() / task_results.shape[0])
+        return torch.stack(coverages)
```
Comment (davidegraff) [inline chemprop/uncertainty/evaluator.py:101]: you should use broadcasting where possible:
```suggestion
         bounds = torch.tensor([-1 / 2, 1 / 2], device=mask.device)
         interval = uncs.unsqueeze(0) * bounds.view([-1] + [1] * preds.ndim)
         lower, upper = preds.unsqueeze(0) + interval
         covered_mask = torch.logical_and(lower <= targets, targets <= upper)
         
         return (covered_mask & mask).sum(0) / mask.sum(0)
```

For context:
```pycon
preds, targets, uncs = torch.randn(3, 128, 4).unbind(0)
mask = torch.randn_like(preds) < 0.8
```

```python
%%timeit
coverages = []
bounds = torch.tensor([-1 / 2, 1 / 2], device=mask.device).view(-1, 1)
for j in range(uncs.shape[1]):
    mask_j = mask[:, j]
    preds_j = preds[:, j][mask_j]
    targets_j = targets[:, j][mask_j]
    interval_j = uncs[:, j][mask_j]
    interval_bounds = bounds * interval_j.unsqueeze(0)
    pred_bounds = preds_j.unsqueeze(0) + interval_bounds
    task_results = torch.logical_and(
        pred_bounds[0, :] <= targets_j, targets_j <= pred_bounds[1, :]
    )
    coverages.append(task_results.sum() / task_results.shape[0])
res1 = torch.stack(coverages)
# 162 µs ± 6.53 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```
vs.
```python
%%timeit
bounds = torch.tensor([-1 / 2, 1 / 2], device=mask.device)
interval = uncs.unsqueeze(0) * bounds.view([2] + [1] * preds.ndim)
lower, upper = preds.unsqueeze(0) + interval
covered_mask = torch.logical_and(lower <= targets, targets <= upper)
res2 =(covered_mask & mask).sum(0) / mask.sum(0)
# 20.8 µs ± 61 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```
and
```python
res1.equal(res2)
# True
```

**Review by davidegraff — COMMENTED** [review]
I leave the rest of the broadcasting work to you

**Review by jonwzheng — CHANGES_REQUESTED** [review]
please also check the docstring for `uncs`:
> uncs (Tensor) – the predicted uncertainties **(variance)** of the shape of n x t

**Review by jonwzheng — APPROVED** [review]
LGTM

**Pull request merged**